lib/compiler.js

```diff
@@ -41,12 +47,19 @@ function scan (text) {
 
   for (; i < len; i++) {
     ch = text.charAt(i)
-    if (TERMINALS[ch]) {
+    if (ch === ESCAPE_CHAR && !escape) {
+      escape = true
+      continue
```

Comment:
I think it's just confusing, using a language within the language itself.

Mask `a\\g` would be written in JS as `a\\\\g`, if you would use JS string `a\\g` it would come out as `a\g` which means that you are trying to escape character `g`. Maybe I should keep grammar for the EscapeSeq as `"\" ? any visible char ?` to not confuse the user.

IMO double escaping is correct and intended behaviour, where as using single backslash within the mask `\` (in JS `"\\"` should fail with parse error as invalid mask `\`.

Reply:
Oh, you’re right: this is very confusing. What are your thoughts on padding for escaping terminals? (e.g. “a//b” and “a((b” to get a “a/b” and “a(b” keys)

Reply:
That might be also an option. I was thinking about quoting the names that contains terminals to get a literal:

`"'a/b/c'(a,b/c)"`

We could also borrow grammar for identifiers from [JMESPath](https://jmespath.org/specification.html#identifiers) or [JSONPath](https://goessner.net/articles/JsonPath/), but that is much more complex.

Solution I've provided is not perfect, but you should be able to escape every character needed.

Reply:
> Mask `a\\g` would be written in JS as `a\\\\g`, if you would use JS string `a\\g` it would come out as `a\g` which means that you are trying to escape character `g`. Maybe I should keep grammar for the EscapeSeq as `"\" ? any visible char ?` to not confuse the user.

Wait hold on, explain this bit to me again, please @mochja. What I was saying is that I expect the test below to pass. It succeeds on the `main` branch, but fails on `mochja:master`:

```js
// in index-test.js
{
  m: '\\',
  o: { '\\': 1 },
  e: { '\\': 1 }
}
```

Reply:
This can't pass as the syntax is invalid as I stated before.

> IMO double escaping is correct and intended behaviour, where as using single backslash within the mask \ (in JS "\\\\") should fail with parse error as invalid mask `\`.

Thats because escape sequence is defined as a pair of the backslash and a character. This test you wrote should fail with a parse error for the mask as a character is always expected after `\`, it also works on main because grammar is different and does not count for escaping sequence. Escape sequence allows you to escape all characters, but it also needs to escape itself, this is the case where you are escaping the escape character.

In JS it behaves the same way, you must have a character after backslash
![image](https://user-images.githubusercontent.com/1218946/167618684-34827dbd-d846-41b5-989b-d2813083ccb0.png)